vendor/github.com/jacobsa/gcloud/gcs/conn.go

```diff
@@ -184,20 +185,32 @@ func (c *conn) OpenBucket(
 	// bucket.
 	_, err = b.ListObjects(ctx, &ListObjectsRequest{MaxResults: 1})
 
-	if typed, ok := err.(*googleapi.Error); ok {
-		switch typed.Code {
-		case http.StatusForbidden:
+	//if typed, ok := err.(*googleapi.Error); ok {
+	//	switch typed.Code {
+	//	case http.StatusForbidden:
+	//		err = fmt.Errorf(
+	//			"Bad credentials for bucket %q. Check the bucket name and your "+
+	//				"credentials.",
+	//			b.Name())
+	//
+	//		return
+	//
+	//	case http.StatusNotFound:
+	//		err = fmt.Errorf("Unknown bucket %q", b.Name())
+	//		return
+	//	}
+	//}
+
+	if err != nil {
+		if strings.Contains(err.Error(), "Error 403") {
```

Comment:
as discussed please dont use string comparision for error. cast it appropriately like this and compare the status code.
(err.(*fmt.wrapError).err).(*googleapi.Error).Code

Reply:
Also check if we can cast error as mentioned in the link Prince shared: https://github.com/jacobsa/gcloud/issues/15